services/jenkins/jenkins-common.js

```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const Joi = require('joi')
+
+const queryParamSchema = Joi.object({
+  disableStrictSSL: Joi.equal(''),
+}).required()
+
+const buildUrl = ({ protocol, host, job, lastBuild = true, plugin }) => {
+  const jobPrefix = job.indexOf('/') > -1 ? '' : 'job/'
+  return `${protocol}://${host}/${jobPrefix}${job}/${
+    lastBuild ? 'lastBuild/' : ''
+  }${plugin ? `${plugin}/` : ''}api/json`
+}
```

Comment:
I'm setting this function up to be re-used in all the other Jenkins services (Tests, etc.) which is why it contains things not explicitly used for the Build service badge